src/routes/(app)/admin[[hash=admin_hash]]/product/[id]/+page.server.ts

```diff
@@ -205,18 +205,18 @@ export const actions: Actions = {
 					...(parsed.restrictPaymentMethods && {
 						paymentMethods: parsed.paymentMethods ?? []
 					}),
-					...(parsed.standalone && { hasVariations: parsed.hasVariations }),
+					...(parsed.standalone && {
+						hasVariations: Object.entries(cleanedVariationLabels?.names || []).length !== 0
+					}),
 					...(parsed.standalone &&
 						parsed.hasVariations && {
```

Comment:
the issue is there.

Do you want to use the `hasVariations` from the frontend or not?

If yes, it should be included the first test.

If not, it should not be preset in the second test, and we should set `const hasVariations = ....` and reuse it for setting `hasVariations`, `variationLabels` and `variationsè

Reply:
No i don't want to use ` hasVariation` because it depends on `variationLabels` is empty or not. And if we uncheck it in the frontend `variationLabels` become empty